filebeat/crawler/prospector.go

```diff
@@ -18,6 +18,8 @@ type Prospector struct {
 	harvesterChan    chan *input.FileEvent
 	registrar        *Registrar
 	done             chan struct{}
+	harvesterStates  []input.FileState
+	stateMutex       sync.Mutex
```

Comment:
The reason I moved it to the prospector level is because it simplified the code. I want to do filtering at the prospector level so all events are already sent through the prospector. As now state is reported with the events, this was the easy place to hook in. I actually think not necessarly all prospectors will use the state concept, but can if needed, so I think having it on the prospector level is a good thing.